src/features/hotspots/root/HotspotsView.tsx

```diff
@@ -105,66 +154,60 @@ const HotspotsView = ({
   }, [hasHotspots, networkHotspots])
 
   useEffect(() => {
-    if (listIsDismissed && !selectedHotspot) {
+    if (viewState === 'list' && !selectedHotspot) {
       if (ownedHotspots && ownedHotspots.length > 0) {
         setSelectedHotspot(ownedHotspots[0])
       } else {
         focusClosestHotspot()
       }
     }
     // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [listIsDismissed, networkHotspots])
-
-  useEffect(() => {
-    if (startOnMap) {
-      detailsRef.current?.present()
-    } else {
-      listRef.current?.present()
-    }
-  }, [dispatch, hasHotspots, startOnMap])
+  }, [viewState, networkHotspots])
 
   const handleLayoutList = useCallback((event: LayoutChangeEvent) => {
     setListHeight(event.nativeEvent.layout.height - 166)
   }, [])
 
-  const handlePresentDetails = useCallback((hotspot: Hotspot) => {
-    setSelectedHotspot(hotspot)
-    setDetailsSnapIndex(1)
-    listRef.current?.dismiss()
-  }, [])
+  const handlePresentDetails = useCallback(
+    (backView: ViewState) => (hotspot: Hotspot) => {
+      updateViewState('details', backView)
+      setSelectedHotspot(hotspot)
+    },
+    [updateViewState],
+  )
 
-  const handleDismissList = useCallback(() => {
-    detailsRef.current?.present()
-    setListIsDismissed(true)
-  }, [])
+  const handleListDismiss = useCallback(() => {
+    if (viewState === 'details') return
+    updateViewState('map')
```

Comment:
I notice now that when I just swipe down the list card, I end up on the map with my last selected hotspot but the details card never shows.

Reply:
We decided to change it up so that it doesn't auto-select a hotspot for you.